internal/client.go

```diff
@@ -626,6 +630,515 @@ type (
+		CreatedAt time.Time
+
+		// LastUpdateAt - When a schedule was last updated
+		LastUpdateAt time.Time
+	}
+
+	// ScheduleDescribeResponse describes the current Schedule details from ScheduleHandle.Describe.
+	ScheduleDescribeResponse struct {
+		// Schedule - Describes the modifiable fields of a schedule.
+		Schedule Schedule
+
+		// Info - Extra information about the schedule.
+		Info ScheduleInfo
+
+		// Memo - Non-indexed user supplied information.
+		Memo *commonpb.Memo
+
+		// SearchAttributes - Indexed info that can be used in query of List schedules APIs (only
+		// supported when Temporal server is using ElasticSearch). The key and value type must be registered on Temporal server side.
+		// Use GetSearchAttributes API to get valid key and corresponding value type.
+		SearchAttributes *commonpb.SearchAttributes
+	}
+
+	// SchedulePolicies describes the current polcies of a schedule.
+	SchedulePolicies struct {
+		// Overlap - Controls what happens when an Action would be started by a Schedule at the same time that an older Action is still
+		// running.
+		Overlap enumspb.ScheduleOverlapPolicy
+
+		// CatchupWindow - The Temporal Server might be down or unavailable at the time when a Schedule should take an Action. When the Server
+		// comes back up, CatchupWindow controls which missed Actions should be taken at that point.
+		CatchupWindow time.Duration
+
+		// PauseOnFailure - When an Action times out or reaches the end of its Retry Policy.
+		PauseOnFailure bool
+	}
+
+	// ScheduleState describes the current state of a schedule.
+	ScheduleState struct {
+		// Note - Informative human-readable message with contextual notes, e.g. the reason
+		// a Schedule is paused. The system may overwrite this message on certain
+		// conditions, e.g. when pause-on-failure happens.
+		Note string
+
+		// Paused - True if the schedule is paused.
+		Paused bool
+
+		// LimitedActions - While true RemainingActions will be decremented for each action taken.
+		// Skipped actions (due to overlap policy) do not count against remaining actions.
+		LimitedActions bool
+
+		// RemainingActions - The Actions remaining in this Schedule. Once this number hits 0, no further Actions are taken.
+		// manual actions through backfill or ScheduleHandle.Trigger still run.
+		RemainingActions int64
+	}
+
+	// Schedule describes a created schedule.
+	Schedule struct {
+		// Action - Which Action to take
+		Action ScheduleAction
+
+		// Schedule - Describes when Actions should be taken.
+		Spec *ScheduleSpec
+
+		// SchedulePolicies - this schedules policies
+		Policy *SchedulePolicies
+
+		// State - this schedules state
+		State *ScheduleState
+	}
+
+	// ScheduleUpdate describes the desired new schedule from ScheduleHandle.Update.
+	ScheduleUpdate struct {
+		// Schedule - New schedule to replace the existing schedule with
+		Schedule *Schedule
+	}
+
+	// ScheduleUpdateOptions configure the parameters for updating a schedule.
+	ScheduleUpdateOptions struct{}
```

Comment:
This is empty right now because the server does not return a failure on conflict token mismatch. Once that is fixed we can add an option to fail on that case and retry.